File: src/baseDatos/Deserializador.java
package baseDatos;

import gestorAplicacion.enrutadorHFC.*;

import java.io.FileInputStream;
import java.io.ObjectInputStream;
import java.io.*;
import java.util.*;

public class Deserializador{

    private static File rutaTemp = new File("src/baseDatos/temp");

    public static void deserializar(){

        File[] docs = rutaTemp.listFiles();
        FileInputStream fIn;
        ObjectInputStream oIn;

        for(File file: docs){
            if(file.getAbsolutePath().contains("Antena")){
                try{
                    fIn = new FileInputStream("src/baseDatos/temp/objetosAntena");
                    oIn = new ObjectInputStream(fIn);
    
                    Antena.setAntenasTotales((ArrayList<Antena>) (oIn.readObject()));
                    oIn.close();
    
                }catch(FileNotFoundException e){
                    e.printStackTrace();

                }catch(IOException e){
                    e.printStackTrace();
                }catch(Exception e){
                    e.printStackTrace();
                }

            }else if(file.getAbsolutePath().contains("Dispositivo")){
                try{
                    fIn = new FileInputStream("src/baseDatos/temp/objetosDispositivo");
                    oIn = new ObjectInputStream(fIn);
    
                    Dispositivo.setDispositivosTotales((ArrayList<Dispositivo>) (oIn.readObject()));
                    oIn.close();
    
                }catch(FileNotFoundException e){
                    e.printStackTrace();
                }catch(IOException e){
                    e.printStackTrace();
                }catch(Exception e){
                    e.printStackTrace();
                }
    
            }else if(file.getAbsolutePath().contains("Servidor")){
                try{
                    fIn = new FileInputStream("src/baseDatos/temp/objetosServidor");
                    oIn = new ObjectInputStream(fIn);
    
                    Servidor.setServidoresTotales((ArrayList<Servidor>)(oIn.readObject()));
                    oIn.close();

                }catch(FileNotFoundException e){
                    e.printStackTrace();
                }catch(IOException e){
                    e.printStackTrace();
                }catch(Exception e){
                    e.printStackTrace();
                }

            }
        } 
    }
}

File: Python/src/baseDatos/deserializador.py
import pickle
import sys
sys.path.append("src/gestorAplicacion/enrutadorHFC")
sys.path.append("src/gestorAplicacion/servicio")

from antena import Antena
from servidor import Servidor
from dispositivo import Dispositivo

class Deserializador:

    @classmethod
    def deserializar(cls):

        Antena.setAntenasTotales(pickle.load(open("src/baseDatos/temp/Antena","rb"))) #NO SE SI PONER EL CLOSE
        Dispositivo.setDispositivosTotales(pickle.load(open("src/baseDatos/temp/Dispositivo","rb")))
        Servidor.setServidoresTotales(pickle.load(open("src/baseDatos/temp/Servidor","rb")))
        